.github/workflows/publish-npmjs-beta.sh

```diff
@@ -0,0 +1,29 @@
+set -x
+
+cd build/packages/js || exit
+
+# Set authentication token for npmjs registry
+npm set //registry.npmjs.org/:_authToken="$NPMJS_TOKEN"
+
+# Set 'beta' suffix in the version starting with beta.0
+sed -i -e 's/-SNAPSHOT"/-beta.0"/g' package.json
+
+# Try to upload the beta version. If it is not available, increase the beta number and try again.
+# Iterate a maximum of MAX_ITERATION times.
+n=1
+MAX_ITERATIONS=30
+while [ $n -le $MAX_ITERATIONS ]; do
```

Comment:
The difference between maven snapshots and npmjs beta versions is that snapshots can be uploaded using the same version, but npmjs beta versions cannot. This is solved by trying to upload starting at beta.0; if it fails, continue with beta.1 and so on.